app/controllers/searches_controller.rb

```diff
@@ -2,23 +2,62 @@ class SearchesController < ApplicationController
   skip_before_action :authenticate_user!
 
   def new
-    @search = Search.new(query: params[:query]) || Search.new
-    @query = []
-    @query << params[:query]
-    @results = ["actor1", "actor2"]
+    set_instance_vars
   end
 
   def create
-    @search = Search.new(search_params)
-    @result = Result.new
-    @search.result = @result
+    query = params[:queries] ? "#{params[:queries]} #{params[:search][:query]}" : params[:search][:query]
+    @search = Search.new(query: query)
     @search.user = current_user if user_signed_in?
-    @search.save ? (redirect_to new_search_path(query: @search.query)) : (render :new)
+    if @search.save
+      redirect_to root_path(construct_query)
```

Comment:
hi i think we should redirect to searches_new becasue root will be the front page now...

Reply:
I share your thought Hannah. Is this possible Gergely ?